include/vikunja/mem/iterator/ConstantIterator.hpp

```diff
@@ -0,0 +1,267 @@
+                 */
+                ConstantIterator(const DataType& value, const IdxType& idx = static_cast<IdxType>(0))
+                    : v(value)
+                    , index(idx)
+                {
+                }
+
+                /**
+                 * @brief Dereference operator to receive the stored value
+                 */
+                NODISCARD ALPAKA_FN_INLINE const DataType& operator*() const
+                {
+                    return v;
+                }
+
+                /**
+                 * @brief Index operator to get stored value at some given offset from this iterator
+                 */
+                NODISCARD ALPAKA_FN_INLINE const DataType& operator[](int) const
+                {
+                    return v;
+                }
+
+#pragma region arithmeticoperators
+                /**
+                 * @brief Postfix increment operator
+                 * @note Use prefix increment operator instead if possible to avoid copies
+                 */
+                NODISCARD ALPAKA_FN_INLINE ConstantIterator operator++()
+                {
+                    ConstantIterator cpy = *this;
+                    ++index;
+                    return cpy;
+                }
+
+                /**
+                 * @brief Prefix increment operator
+                 */
+                NODISCARD ALPAKA_FN_INLINE ConstantIterator& operator++(int)
+                {
+                    ++index;
+                    return *this;
+                }
+
+                /**
+                 * @brief Postfix decrement operator
+                 * @note Use prefix decrement operator instead if possible to avoid copies
+                 */
+                NODISCARD ALPAKA_FN_INLINE ConstantIterator operator--()
+                {
+                    ConstantIterator cpy = *this;
+                    --index;
+                    return cpy;
+                }
+
+                /**
+                 * @brief Prefix decrement operator
+                 */
+                NODISCARD ALPAKA_FN_INLINE ConstantIterator operator--(int)
+                {
+                    --index;
+                    return *this;
+                }
+
+                /**
+                 * @brief Add an index to this iterator
+                 */
+                NODISCARD ALPAKA_FN_INLINE ConstantIterator operator+(const IdxType idx) const
+                {
+                    return ConstantIterator(v, index + idx);
+                }
+
+                /**
+                 * @brief Add a second constant iterator of the same value to this one
+                 */
+                NODISCARD ALPAKA_FN_INLINE ConstantIterator operator+(const ConstantIterator& other) const
+                {
+                    assert(v == other.v && "Can't add constant iterators of different values!");
+                    return ConstantIterator(v, index + other.index);
+                }
```

Comment:
This overload makes no sense for iterators. When do you need to add two iterators? Iterators are conceptually positions in a sequence. You cannot add two positions in a sequence.

Reply:
You're right, I read this https://en.cppreference.com/w/cpp/iterator/random_access_iterator
and got confused because of the b-a. Every addition is only between iterator and index type.
I also have to change the operator- then to return an index, not a new iterator